BrightFutures/Future.swift

```diff
@@ -383,12 +379,7 @@ public extension Future {
     /// Returns self
     public func onSuccess(context c: ExecutionContext = executionContextForCurrentContext(), callback: SuccessCallback) -> Future<T> {
         self.onComplete(context: c) { result in
-            switch result {
-            case .Success(let val):
-                callback(val.value)
-            default:
-                break
-            }
+            result.analysis(ifSuccess: callback, ifFailure: { _ in })
```

Comment:
BrightFutures used to have something like `analysis` but I removed it because I thought it looked strange. But since antitypical/Result uses it, it is probably not such a bad idea and something I could get used to.